securedrop/models.py

```diff
@@ -259,28 +264,29 @@ def __repr__(self):
             self.username,
             " [admin]" if self.is_admin else "")
 
-    def _gen_salt(self, salt_bytes=32):
-        return os.urandom(salt_bytes)
+    _LEGACY_SCRYPT_PARAMS = dict(N=2**14, r=8, p=1)
 
-    _SCRYPT_PARAMS = dict(N=2**14, r=8, p=1)
-
-    def _scrypt_hash(self, password, salt, params=None):
-        if not params:
-            params = self._SCRYPT_PARAMS
-        return scrypt.hash(str(password), salt, **params)
+    def _scrypt_hash(self, password, salt):
+        return scrypt.hash(str(password), salt, **self._LEGACY_SCRYPT_PARAMS)
 
     MAX_PASSWORD_LEN = 128
     MIN_PASSWORD_LEN = 14
 
-    def set_password(self, password):
-        self.check_password_acceptable(password)
+    def set_password(self, passphrase):
+        self.check_password_acceptable(passphrase)
+
+        # "migrate" from the legacy case
+        if not self.passphrase_hash:
+            self.passphrase_hash = \
+                argon2.using(**ARGON2_PARAMS).hash(passphrase)
+            self.pw_hash = None
+            self.pw_salt = None
```

Comment:
setting `pw_salt` to None is such an important implementation detail that it should have a comment to indicate to everyone that in `passlib.hash.argon2` salts are auto-generated and stored in the `passphrase_hash` column

Reply:
Fixed in both place.